sampler.go

```diff
@@ -174,6 +181,177 @@ func (s *rateLimitingSampler) Equal(other Sampler) bool {
 
 // -----------------------
 
+type lowerBoundSampler struct {
+	rateLimitingSampler Sampler
+	samplingRate        float64
+	tags                []Tag
+}
+
+// NewLowerBoundSampler creates a sampler that samples at most maxTracesPerSecond. It is a wrapper around
+// a rateLimitingSampler with different tags since it's intended to be used by the adaptiveSampler.
+// The sampler.param tag is sampling rate rather than maxTracesPerSecond. This is used to tell the collector
+// that the trace sampled by this sampler used the samplingRate to determine the throughput for the operation.
+func NewLowerBoundSampler(maxTracesPerSecond, samplingRate float64) (Sampler, error) {
+	tags := []Tag{
+		{key: SamplerTypeTagKey, value: SamplerTypeLowerBound},
+		{key: SamplerParamTagKey, value: samplingRate},
+	}
+	rateLimitingSampler, err := NewRateLimitingSampler(maxTracesPerSecond)
+	if err != nil {
+		return nil, err
+	}
+	return &lowerBoundSampler{
+		rateLimitingSampler: rateLimitingSampler,
+		samplingRate:        samplingRate,
+		tags:                tags,
+	}, nil
+}
+
+// IsSampled implements IsSampled() of Sampler.
+func (s *lowerBoundSampler) IsSampled(id uint64, operation string) (bool, []Tag) {
+	sampled, _ := s.rateLimitingSampler.IsSampled(id, operation)
+	return sampled, s.tags
+}
+
+func (s *lowerBoundSampler) Close() {
+	s.rateLimitingSampler.Close()
+}
+
+func (s *lowerBoundSampler) Equal(other Sampler) bool {
+	if o, ok := other.(*lowerBoundSampler); ok {
+		return s.samplingRate == o.samplingRate &&
+			s.rateLimitingSampler.Equal(o.rateLimitingSampler)
+	}
+	return false
+}
+
+// -----------------------
+
+// sampler for the operation level
+type operationSampler struct {
+	probabilisticSampler Sampler
+	lowerBoundSampler    Sampler
```

Comment:
can also strongly type - the name already implies the type, so no reason to use generic `Sampler`